function_mapper.hpp

```diff
@@ -0,0 +1,129 @@
+#pragma once
+
+#include <iostream>
+#include <vector>
+
+#include "redis.hpp"
```

Comment:
Instead of including `redis.hpp` here, just include the `store.hpp`. If you check the includes in `redis.hpp`, you'll see that you're including `function_mapper.hpp` there. Doing this creates a circular dependency and it can cause issues. Maybe it worked for you on your machine, but it didn't work using `clang++` compiler on my machine. Plus the editor reported a ton of errors because of this. So I would suggest avoiding this kind of including and try to structure your includes as trees (no circular dependencies) and have them explicitly listed in each file.